app/parse.py

```diff
@@ -8,9 +19,145 @@
     tags: list[str]
 
 
+@dataclass
+class Author:
+    name: str
+    biography: str
+
+
+QUOTE_FIELDS = [field.name for field in fields(Quote)]
+AUTHOR_FIELDS = [field.name for field in fields(Author)]
+authors_list = []
+
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)8s]: %(message)s",
+    handlers=[
+        logging.FileHandler("parser.log"),
+        logging.StreamHandler(sys.stdout),
+    ]
+)
+
+
+def parse_single_quote(quote_soup: Tag) -> Quote:
+    author_name = quote_soup.select_one(".author").text
+    if author_name not in authors_list:
+        authors_list.append(author_name)
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=author_name,
+        tags=quote_soup.select_one(".tags").text.split()[1:]
+    )
+
+
+def get_single_author(name: str, biography: str) -> Author:
+    return Author(
+        name=name,
+        biography=biography,
+    )
+
+
+def get_single_page_quotes(page_soup: Tag) -> [Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote_soup) for quote_soup in quotes]
+
+
+def get_all_pages_quotes() -> [Quote]:
+    # logging
+    time = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+    logging.info(f"{time} - Start parsing quotes")
+
+    page_text = requests.get(BASE_URL).content
+    page_soup = BeautifulSoup(page_text, "html.parser")
+
+    all_quotes = get_single_page_quotes(page_soup)
+
+    # next button searching
+    next_element = page_soup.select_one(".next")
+    while next_element is not None:
+        # Checking the existence of an element and the href attribute
+        if next_element and next_element.a and "href" in next_element.a.attrs:
+            next_href = next_element.a["href"]
+            # Using a Regular Expression to Extract a Number from a URL
+            match = re.search(r"/page/(\d+)/", next_href)
+            if match:
+                next_page_number = match.group(1)
```

Comment:
The method for extracting `next_page_number` relies on a specific URL structure. This could break if the URL changes. Consider implementing a more robust method for extracting the page number.